src/hotspot/share/gc/g1/g1ConcurrentMark.hpp

```diff
@@ -556,6 +560,16 @@ class G1ConcurrentMark : public CHeapObj<mtGC> {
   // Set live bytes for concurrent marking.
   void set_live_bytes(uint region, size_t live_bytes) { _region_mark_stats[region]._live_words = live_bytes / HeapWordSize; }
 
+  // Update the TAMS for the given region to the current top.
+  inline void update_top_at_mark_start(HeapRegion* r);
+  // Reset the TAMS for the given region to bottom of that region.
+  inline void reset_top_at_mark_start(HeapRegion* r);
+
+  inline HeapWord* top_at_mark_start(const HeapRegion* r) const;
+  inline HeapWord* top_at_mark_start(uint region) const;
```

Comment:
Maybe call it `region_index` or sth alike?  (In the impl of these methods, having `region` being the index is a bit confusing, IMO.)